internal/keys/datastore_doc.go

```diff
@@ -0,0 +1,292 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package keys
+
+import (
+	"strconv"
+
+	ds "github.com/ipfs/go-datastore"
+
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/internal/encoding"
+)
+
+// InstanceType is a type that represents the type of instance.
+type InstanceType string
+
+const (
+	// ValueKey is a type that represents a value instance.
+	ValueKey = InstanceType("v")
+	// PriorityKey is a type that represents a priority instance.
+	PriorityKey = InstanceType("p")
+	// DeletedKey is a type that represents a deleted document.
+	DeletedKey = InstanceType("d")
+
+	DATASTORE_DOC_VERSION_FIELD_ID = "v"
+)
+
+// DataStoreKey is a type that represents a key in the database.
+type DataStoreKey struct {
```

Comment:
suggestion: Since the package name is `keys` (should be `key`), the `Key` suffix is less relevant. I would suggest removing it to get `key.DataStore`. This applies for all keys.

Reply:
Yes, I agree, but I didn't want to bother dealing with the many niggles like this already present in the code I moved.  I just want to break it up so I can add new stuff in a clean way.

Reply:
We've been doing this often lately — refactoring but not cleaning things up fully. It feels like a bad habit that's setting in that might hurt us in the future. This one is minor but still part of the habit so I'm pointing it out.

Reply:
Code is never cleaned up fully.  Leaving as is.

Reply:
> We've been doing this often lately — refactoring but not cleaning things up fully. It feels like a bad habit that's setting in that might hurt us in the future.

I am curious to know where we have done this lately? We should def have issues to track any pending cleanups if there were any.